pinot-tools/src/main/resources/conf/log4j2.xml

```diff
@@ -20,33 +20,31 @@
 
 -->
 <Configuration>
-  <Appenders>
-    <Console name="console" target="SYSTEM_OUT">
-      <PatternLayout>
-        <pattern>%d{yyyy/MM/dd HH:mm:ss.SSS} %p [%c{1}] [%t] %m%n</pattern>
-      </PatternLayout>
-    </Console>
-    <RandomAccessFile name="ingestionJobLog" fileName="pinotIngestionJob.log" immediateFlush="false">
-      <PatternLayout>
-        <Pattern>%d{yyyy/MM/dd HH:mm:ss.SSS} %p [%c{1}] [%t] %m%n</Pattern>
-      </PatternLayout>
-    </RandomAccessFile>
 
+  <Properties>
+    <Property name="LOG_ROOT">logs</Property>
+    <Property name="LOG_PATTERN">%d{yyyy/MM/dd HH:mm:ss.SSS} %p [%c{1}] [%t] %m%n</Property>
+    <Property name="PINOT_COMPONENT">all</Property>
+  </Properties>
+  <Appenders>
+    <Console name="console" target="SYSTEM_OUT" />
+    <RollingFile name="pinotLog" fileName="${env:LOG_ROOT}/pinot-${env:PINOT_COMPONENT}.log"
+                 filePattern="${env:LOG_ROOT}/pinot-${env:PINOT_COMPONENT}-%d{yyyy-MM-dd}-%i.log" immediateFlush="false">
```

Comment:
Few comments:

1. Log file name need not have date pattern since it is only size based rollover
2. Enable compression to reduce the space used and increase the amount of logs to back in date. Use bzip2 compression since it has `bzgrep` tool to grep logs without uncompressing them.
3. Follow the convention of *NIX rolled logs by moving `%i ` after `.log` extension

Suggested filePattern that takes care of #1 and #2:

`filePattern="${env:LOG_ROOT}/pinot-${env:PINOT_COMPONENT}.log.%i.bz2"`

Reply:
Sounds good! I feel it's still better to use gzip for log files, which can be open by vim and grep by zgrep also. Changed to `${env:LOG_ROOT}/pinot-${env:PINOT_COMPONENT}.log.%i.gz`